### PR TITLE
Package pipeline_spec proto into SDK

### PR DESCRIPTION
**Pull request by chensun**
- Fork the pipeline_spec proto added by #4371 to SDK folder
- Update the type of the `ImporterSpec.artifact_uri` from `string` to `ValueOrRuntimeParameter`
- Add protoc generated file and `__init__.py`

/cc @numerology @neuromage @Ark-kun @hongye-sun

### CODE REVIEW

File: sdk/python/kfp/proto/pipeline_spec.proto [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+      Exec pre_cache_check = 1;
+    }
+    // The lifecycle hooks of the container executor.
+    Lifecycle lifecycle = 4;
+  }
+
+  // The specification to import or reimport a new artifact to the pipeline.
+  message ImporterSpec {
+    // The URI of the artifact.
+    ValueOrRuntimeParameter artifact_uri = 1;
```
Comment (chensun) [inline sdk/python/kfp/proto/pipeline_spec.proto:259]: @hongye-sun please review this type change is correct.

Comment (chensun) [reply sdk/python/kfp/proto/pipeline_spec.proto:259]: confirmed

**Review by numerology — COMMENTED** [review]
Nit suggestion: add copyright statement in all *.py. Otherwise LGTM :)

**Comment by chensun** [comment]
> Nit suggestion: add copyright statement in all *.py. Otherwise LGTM :)

Done

**Comment by numerology** [comment]
Thanks Chen!

**Pull request merged**